apps/cms.c

```diff
@@ -301,7 +301,7 @@ int cms_main(int argc, char **argv)
     OPTION_CHOICE o;
     OSSL_LIB_CTX *libctx = app_get0_libctx();
 
-    if ((vpm = X509_VERIFY_PARAM_new()) == NULL)
+    if (encerts == NULL || vpm == NULL)
```

Comment:
Use goto end; otherwise it theoretically can leak one of those.

Reply:
Ah, right; done.
And rebased to fix merge conflicts.